Do not die if SSL certificates aren't present

If a SSL certificate is not present, the proxy fails to start with a weird message:
```
$ GO_PROXY_CERT_FILE=no_a_real_file.txt go-proxy
2018/03/27 09:50:11 Starting proxy server...
2018/03/27 09:50:11 Starting admin server...
Starting proxy at: https://localhost:33443
panic: open no_a_real_file.txt: no such file or directory

goroutine 1 [running]:
main.startProxy()
	~/go-proxy/main.go:27 +0x92
main.main()
	~/go-proxy/main.go:12 +0x3a
```

It would be a better experience if a message was logged explaining why HTTPS is not available and start the server in HTTP mode.

--- a/proxy/http_proxy.go
+++ b/proxy/http_proxy.go
@@ -14,8 +14,6 @@ import (
 	myhttp "github.com/visola/go-proxy/http"
 )
 
-var urlToProxy = fmt.Sprintf("https://localhost:%d", port)
-
 func proxyRequest(req *http.Request, w http.ResponseWriter, match *config.MatchResult) (*proxyResponse, error) {
 	mapping := match.Mapping
 	newPath := mapping.To + "/" + match.NewPath[len(mapping.From):]
@@ -85,7 +83,7 @@ func proxyRequest(req *http.Request, w http.ResponseWriter, match *config.MatchR
 			// Fix location headers to point to proxy
 			if strings.ToLower(name) == "location" {
 				if strings.HasPrefix(value, mapping.To) {
-					value = urlToProxy + value[len(mapping.To):]
+					value = GetURL() + value[len(mapping.To):]
 				}
 			}
 			w.Header().Add(name, value)

--- a/proxy/server.go
+++ b/proxy/server.go
@@ -13,8 +13,6 @@ import (
 	"github.com/visola/go-proxy/statistics"
 )
 
-var port = 33443
-
 // StartProxyServer starts the proxy server
 func StartProxyServer() error {
 	certFile := os.Getenv("GO_PROXY_CERT_FILE")
@@ -23,9 +21,14 @@ func StartProxyServer() error {
 	proxyServer := http.NewServeMux()
 	proxyServer.HandleFunc("/", requestHandler)
 
-	fmt.Printf("Starting proxy at: https://localhost:%d\n", port)
-	bindError := http.ListenAndServeTLS(fmt.Sprintf(":%d", port), certFile, keyFile, proxyServer)
-	return bindError
+	if certFile == "" || keyFile == "" {
+		fmt.Printf("Starting proxy at: %s\n", GetURL())
+		return http.ListenAndServe(fmt.Sprintf(":%d", port), proxyServer)
+	}
+
+	isSSL = true
+	fmt.Printf("Starting proxy at: %s\n", GetURL())
+	return http.ListenAndServeTLS(fmt.Sprintf(":%d", port), certFile, keyFile, proxyServer)
 }
 
 func requestHandler(w http.ResponseWriter, req *http.Request) {

--- a/proxy/server_url.go
+++ b/proxy/server_url.go
@@ -1,0 +1,16 @@
+package proxy
+
+import "fmt"
+
+var port = 33443
+var isSSL = false
+
+// GetURL returns the full URL to access the proxy server
+func GetURL() string {
+	protocol := "http"
+	if isSSL {
+		protocol = "https"
+	}
+
+	return fmt.Sprintf("%s://%s:%d", protocol, "localhost", port)
+}

